### PR TITLE
feat: apply themev2 for channel list

### PR DESCRIPTION
**Pull request by szuperaz**
Changes in stream-chat-css: https://github.com/GetStream/stream-chat-css/pull/83

### CODE REVIEW

File: projects/sample-app/src/app/app.component.scss [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#root {
```
Comment (szuperaz) [inline projects/sample-app/src/app/app.component.scss:1]: chat UI layout controlled by integrator (not by stream-chat-css)

File: docusaurus/docs/Angular/concepts/themeing-and-css-v2.mdx [hunk 2]
```diff
@@ -0,0 +1,291 @@
+---
+id: themeingv2
+title: Themeing and CSS v2
+---
+
+import ChannelPreviewThemeScreenshot1 from "../assets/channel-preview-theme1-screenshot.png";
```
Comment (szuperaz) [inline docusaurus/docs/Angular/concepts/themeing-and-css-v2.mdx:6]: all screenshots will be updated once theming is done

File: docusaurus/docs/Angular/concepts/themeing-and-css-v2.mdx [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+#### Theme variables
+
+You can use theme layer variables to change the look and feels for the whole chat UI (as opposed to changing the design of individual components), these variables can be grouped into three categories:
+
+- Colors
+- Fonts
+- Border radiuses
+- Spacing (makes the chat UI more spacious/compact)
+
+The full list of theme variables (TODO: provide a better list with descriptions)
```
Comment (szuperaz) [inline docusaurus/docs/Angular/concepts/themeing-and-css-v2.mdx:53]: I'd like to research if there is a way to generate docs for SCSS variables based on doc comments

File: projects/stream-chat-angular/src/lib/channel-preview/channel-preview.component.html [hunk 4]
```diff
@@ -9,13 +10,17 @@
     <stream-avatar-placeholder
       name="{{ avatarName }}"
       imageUrl="{{ avatarImage }}"
-      [size]="40"
       type="channel"
       [channel]="channel"
       location="channel-preview"
+      [size]="49"
```
Comment (szuperaz) [inline projects/stream-chat-angular/src/lib/channel-preview/channel-preview.component.html:16]: Avatar size is controlled in HTML code, theme v1 uses 40px, v2 uses 49px -> I couldn't find an easy solution for using different values for the different themes, screenshots attached for the changes in theme v1, it's annoying to me, but I couldn't find an easy solution
Screenshot 2022-05-06 at 14 48 36
Screenshot 2022-05-06 at 14 51 55

File: docusaurus/docs/Angular/concepts/themeing-and-css-v2.mdx [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+
+or if you're using CSS:
+
+```css
+@import "../../stream-chat-angular/src/assets/styles/v2/css/index.layout.css";
+```
+
+<img src={ChannelListLayoutScreenshot} width="500" />
+
+The result will be a minimalistic UI without
```
Comment (szuperaz) [inline docusaurus/docs/Angular/concepts/themeing-and-css-v2.mdx:209]: @petyosi let me know if you had a more minimalistic UI here, it was hard for me to decide which should be part of the layout

File: projects/sample-app/src/app/app.component.ts [hunk 6]
```diff
@@ -27,5 +29,6 @@ export class AppComponent {
       members: { $in: [environment.userId] },
     });
     this.streamI18nService.setTranslation();
+    setTimeout(() => themeService.theme$.next('dark'), 5000);
```
Comment (petyosi) [inline projects/sample-app/src/app/app.component.ts:32]: :D probably a debug leftover?

File: projects/stream-chat-angular/src/lib/channel-preview/channel-preview.component.html (hunk 4 above)
Comment (petyosi) [reply projects/stream-chat-angular/src/lib/channel-preview/channel-preview.component.html:16]: Let's note this down for later consideration. I don't like sizing on component level as well, but not the end of the world.

File: projects/sample-app/src/app/app.component.ts (hunk 6 above)
Comment (szuperaz) [reply projects/sample-app/src/app/app.component.ts:32]: Thanks

File: projects/stream-chat-angular/src/lib/channel-list/channel-list.component.ts [hunk 7]
```diff
@@ -30,14 +31,17 @@ export class ChannelListComponent implements AfterViewInit, OnDestroy {
   isOpen$: Observable<boolean>;
   hasMoreChannels$: Observable<boolean>;
   customChannelPreviewTemplate: TemplateRef<ChannelPreviewContext> | undefined;
+  theme$: Observable<'light' | 'dark'>;
```
[the remark this answers is not in the document]
Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/channel-list/channel-list.component.ts:34]: Added support for this, and documentation

File: projects/stream-chat-angular/src/lib/channel-preview/channel-preview.component.html (hunk 4 above)
Comment (szuperaz) [reply projects/stream-chat-angular/src/lib/channel-preview/channel-preview.component.html:16]: Issue created for it: https://github.com/GetStream/stream-chat-css/issues/84

**Pull request merged**